backend/api/views/task_profiling.py

```diff
@@ -53,14 +55,31 @@ def perform_update(self, serializer):
         return serializer.save(**kwargs)
 
 
-class TaskProfilingStepViewSet(mixins.CreateModelMixin, GenericViewSet):
+class TaskProfilingStepViewSet(mixins.CreateModelMixin, mixins.UpdateModelMixin, GenericViewSet):
     serializer_class = ProfilingStepSerializer
     authentication_classes = api_settings.DEFAULT_AUTHENTICATION_CLASSES + [BasicAuthentication]
     permission_classes = [IsAuthenticated, IsCurrentBackendOrReadOnly]
+    lookup_field = "step"
+
+    def get_queryset(self) -> QuerySet[TaskProfiling]:
+        return ProfilingStep.objects.filter(compute_task_profile__compute_task__channel=get_channel_name(self.request))
 
     def create(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         task_profile_pk = kwargs["task_profiling_pk"]
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
-        serializer.save(compute_task_profile_id=task_profile_pk)
-        return Response(serializer.data)
+        try:
+            serializer.save(compute_task_profile_id=task_profile_pk)
+        except IntegrityError:
+            data = {"detail": f"Profiling with key {task_profile_pk} and name {request.data['step']} already exists"}
+            return Response(data, status=status.HTTP_409_CONFLICT)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def get_object(self):
+        queryset = self.get_queryset()
+        filters = {**self.kwargs}
+        filters["compute_task_profile_id"] = filters.pop("task_profiling_pk")
+        obj = get_object_or_404(queryset, **filters)
```

Comment:
we are not catching the eventual 404 exception?

Reply:
No, this is handled by Django, DRF. Same behavior than `get_object` in `GenericAPIView` which is used by other viewsets.